packages/pagination/src/index.ts

```diff
@@ -94,201 +109,213 @@ export default defineComponent({
+        }),
+        slot: slots?.default?.() ?? null,
+        total: h(Total, { total: isAbsent(props.total) ? 0 : props.total }),
+      }
 
-      getValidCurrentPage,
-      emitChange,
-      handleCurrentChange,
+      const components = props.layout.split(',').map((item: string) => item.trim())
 
-      prev,
-      next,
-    }
-  },
-  render() {
-    const layout = this.layout
-
-    if (!layout) return null
-    if (this.hideOnSinglePage && this.internalPageCount <= 1) return null
-
-    const rootNode = h('div', {
-      class: [
-        'el-pagination',
-        {
-          'is-background': this.background,
-          'el-pagination--small': this.small,
-        },
-      ],
-    })
-    const rootChildren = []
-    const rightWrapperChildren = []
-    const rightWrapperRoot = h('div', { class: 'el-pagination__rightwrapper' }, rightWrapperChildren)
-    const TEMPLATE_MAP = {
-      prev: h(Prev, {
-        disabled: this.disabled,
-        currentPage: this.internalCurrentPage,
-        prevText: this.prevText,
-        onClick: this.prev,
-      }),
-      jumper: h(Jumper),
-      pager: h(Pager, {
-        currentPage: this.internalCurrentPage,
-        pageCount: this.internalPageCount,
-        pagerCount: this.pagerCount,
-        onChange: this.handleCurrentChange,
-        disabled: this.disabled,
-      }),
-      next: h(Next, {
-        disabled: this.disabled,
-        currentPage: this.internalCurrentPage,
-        pageCount: this.internalPageCount,
-        nextText: this.nextText,
-        onClick: this.next,
-      }),
-      sizes: h(Sizes, {
-        pageSize: this.pageSize,
-        pageSizes: this.pageSizes,
-        popperClass: this.popperClass,
-        disabled: this.disabled,
-      }),
-      slot: this.$slots?.default?.() ?? null,
-      total: h(Total, { total: this.total }),
-    }
+      let haveRightWrapper = false
 
-    const components = layout.split(',').map((item: string) => item.trim())
-
-    let haveRightWrapper = false
+      components.forEach((c: keyof typeof TEMPLATE_MAP | '->') => {
+        if (c === '->') {
+          haveRightWrapper = true
+          return
+        }
+        if (!haveRightWrapper) {
+          rootChildren.push(TEMPLATE_MAP[c])
+        } else {
+          rightWrapperChildren.push(TEMPLATE_MAP[c])
+        }
+      })
 
-    components.forEach((c: keyof typeof TEMPLATE_MAP | '->') => {
-      if (c === '->') {
-        haveRightWrapper = true
-        return
-      }
-      if (!haveRightWrapper) {
-        rootChildren.push(TEMPLATE_MAP[c])
-      } else {
-        rightWrapperChildren.push(TEMPLATE_MAP[c])
+      if (haveRightWrapper && rightWrapperChildren.length > 0) {
+        rootChildren.unshift(rightWrapperRoot)
       }
-    })
 
-    if (haveRightWrapper && rightWrapperChildren.length > 0) {
-      rootChildren.unshift(rightWrapperRoot)
+      return h('div', {
+        role: 'pagination',
+        ['aria-label']: 'pagination',
```

Comment:
Why do we use `[]` to wrap the key here?

Reply:
My stupid misunderstanding